internal/controller/operator/deployment.go

```diff
@@ -1032,4 +1107,42 @@ func buildIdpVolumes(ldapCACert string, routerCertSecret string) []corev1.Volume
 			},
 		},
 	}
+	if len(auditSecretName) > 0 {
+		auditVolume := corev1.Volume{
+			Name: IMAuditTLSVolume,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: auditSecretName,
+					Items: []corev1.KeyToPath{
+						{
+							Key:  "tls.crt",
+							Path: "tls.crt",
+						},
+						{
+							Key:  "tls.key",
+							Path: "tls.key",
+						},
+						{
+							Key:  "ca.crt",
+							Path: "ca.crt",
+						},
+					},
+					DefaultMode: &partialAccess,
+				},
+			},
+		}
+		volumes = EnsureVolumePresent(volumes, auditVolume)
```

Comment:
I don't think this extra function is needed; it should be possible with just an `append`. This function is used to build what the Operator thinks the Deployment should contain, so there's no threat of infinitely appending redundant volumes here.

Reply:
fixed

Reply:
It still appears to be here?